diff --git a/src/wl/sys/wl_linux.c b/src/wl/sys/wl_linux.c
index 75cea23..92d70f7 100644
--- a/src/wl/sys/wl_linux.c
+++ b/src/wl/sys/wl_linux.c
@@ -1655,7 +1655,11 @@ wl_ioctl(struct net_device *dev, struct ifreq *ifr, int cmd)
 		goto done2;
 	}
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 9, 0)
+	if (uaccess_kernel())
+#else
 	if (segment_eq(get_fs(), KERNEL_DS))
+#endif
 		buf = ioc.buf;
 
 	else if (ioc.buf) {
